### PR TITLE
Feature/site search

### PR DESCRIPTION
**Pull request by TrebBrennan**
screen-shot-2016-10-25-at-12 11 03-pm

A plug-n-play site-search component designed to be as modular and responsive as possible.
- Values for colours etc are all declared at the top of the partial so it's easy to customise as needed.
- Checked in a variety of browsers but ie7 needs some love.
- Validated HTML for W3C and WCAG.
- Accessibility discussed and approved with SME.

PS... Sorry, not-sorry, for the jQuery :P

### CODE REVIEW

**Comment by klepas** [comment]
Things that I see left here:
- possible design changes to make the input field match the existing input design (ez)
- rewrite the jquery component in plain JS (Petch? :S)
- incorporate into our codebase
- final testing

**Comment by klepas** [comment]
@petronbot I also wanna tweak the vars for the input style so that it matches our other text input fields, just for consistency.

**Comment by klepas** [comment]
@petronbot I reckon you’ll have to get @danwhitmarsh to review your eidts; I can’t. :S

File: assets/sass/_components.scss [hunk 1]
```diff
@@ -8,8 +8,8 @@
 @import 'components/navigation';
```
Comment (klepas) [inline assets/sass/_components.scss:8]: @petronbot Just saw this: are we sure we won’t run into any cascading issues wrt. vars or other things that are introduced in a partial that now is imported after a partial that might rely or call it?]

**Comment by joolswood** [comment]
@petronbot I can take a look at this after AusDTO/dto-design-guide#25

**Comment by gordcorp** [comment]
FYI have noticed a slight glitch where this is implemented on the dta-website. If you do a search, and then hit the back button, the search query and placeholder text are drawn on top of it. 

image

**Comment by klepas** [comment]
Found another item to hunt down — this is what happens when JS is disabled:

screen shot 2016-11-07 at 1 50 26 pm

**Comment by TrebBrennan** [comment]
@klepas just set the variable on this line:
https://github.com/AusDTO/gov-au-ui-kit/pull/432/files#diff-964067dbbd2240240a0ac21c855c873cR67

RE: @bjgordon's bug above, the JS needs to also be triggered onload not just input change.

File: assets/js/components/site-search.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  _addEventListener(siteSearchInput, formChangeEvents, _onFormChange);
+}
+
+function _onFormChange(event) {
+  let element = event.target;
+  element.classList.toggle();
+}
+
+function _addEventListener(elem, triggers, callback) {
+  if (typeof window.addEventListener === undefined) {
```
Comment (nathf) [inline assets/js/components/site-search.js:17]: `undefined` here should be a string? And shouldn't this `if` be inverse?

**Comment by joolswood** [comment]
We don't have any doco on the component yet @hannah-ustwo suggest reopening or creating a new ticket.

**Pull request closed without merging**